src/arrayMethodSort.js

```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const defaultCompare = function(a, b) {
+      switch (true) {
+        case (`${a}` > `${b}`): return 1;
+        case (`${a}` < `${b}`): return -1;
+        case (`${a}` === `${b}`): return 0;
+        default: throw new Error('Error: defaulCompre');
+      }
+    };
+    const comparator = compareFunction !== undefined
+      ? compareFunction
+      : defaultCompare;
+    let isSorted;
+
+    do {
+      isSorted = true;
+
+      for (let i = 1; i < this.length; i++) {
+        const previous = this[i - 1];
```

Comment:
can you use the array method instead of a simple loop?

Reply:
I tried to implement .forEach method here, but that solution looks complicated and less readable to me.